lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1041,10 +1042,11 @@ def test_parameter_versions(self):
                          % (pv['param'], pv['min_version'])))
 
         for publish_def in self.client.module.params.get('publish', []):
-            if ('mode' in publish_def.keys() and
-                    (LooseVersion(self.client.version()['ApiVersion']) <
-                     LooseVersion('1.25'))):
-                self.client.module.fail_json(msg='publish.mode parameter supported only with api_version>=1.25')
+            if 'mode' in publish_def.keys():
+                if LooseVersion(self.client.version()['ApiVersion']) < LooseVersion('1.25'):
+                    self.client.module.fail_json(msg='publish.mode parameter supported only with api_version>=1.25')
+                if not HAS_DOCKER_PY_3:
```

Comment:
I think it's better to use `LooseVersion(docker_version) < LooseVersion('3.0.0')`, to reduce the amount of code which has to be adjusted once docker-py 4.0.0 comes out :)
(For that, you need to `import docker_version from module_utils.docker_common`.)

Reply:
Yes, it's better that way